src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -704,4 +747,81 @@ private RepairRun buildRepairRunFromRow(Row repairRunResult, UUID id){
         .build(id);
   }
 
+  @Override
+  public boolean takeLeadOnSegment(UUID segmentId) {
+    LOG.debug("Trying to take lead on segment {}", segmentId);
+    ResultSet lwtResult = session.execute(getLeadOnSegmentPrepStmt.bind(segmentId, ReaperApplication.REAPER_INSTANCE_ID, ReaperApplication.getInstanceAddress()));
+    if (lwtResult.wasApplied()) {
+      LOG.debug("Took lead on segment {}", segmentId);
+      return true;
+    }
+    
+    // Another instance took the lead on the segment
+    LOG.debug("Could not take lead on segment {}", segmentId);
+    return false;
+  }
+
+  @Override
+  public boolean renewLeadOnSegment(UUID segmentId) {
+    ResultSet lwtResult = session.execute(renewLeadOnSegmentPrepStmt.bind(ReaperApplication.REAPER_INSTANCE_ID, ReaperApplication.getInstanceAddress(), segmentId, ReaperApplication.REAPER_INSTANCE_ID));
+    if (lwtResult.wasApplied()) {
+      LOG.debug("Renewed lead on segment {}", segmentId);
+      return true;
+    }
+    
+    LOG.debug("Failed to renew lead on segment {}", segmentId);
+    return false;
+  }
+
+  @Override
+  public void releaseLeadOnSegment(UUID segmentId) {
+    ResultSet lwtResult = session.execute(releaseLeadOnSegmentPrepStmt.bind(segmentId, ReaperApplication.REAPER_INSTANCE_ID));
+    if (lwtResult.wasApplied()) {
+      LOG.debug("Released lead on segment {}", segmentId);
+    } else {
+      LOG.error("Could not release lead on segment {}", segmentId);
+    }
+  }
+
+  @Override
+  public void storeHostMetrics(HostMetrics hostMetrics) {    
+    session.execute(storeHostMetricsPrepStmt.bind(hostMetrics.getHostAddress(), hostMetrics.getPendingCompactions(), hostMetrics.hasRepairRunning(), hostMetrics.getActiveAnticompactions()));
+  }
+
+  @Override
+  public Optional<HostMetrics> getHostMetrics(String hostName) {
+    ResultSet result = session.execute(getHostMetricsPrepStmt.bind(hostName));
+    for(Row metrics:result) {
+      return Optional.of(HostMetrics.builder().withHostAddress(hostName)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "return" statement or make it conditional. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1751)